File: Problems/611. Valid Triangle Number/mine.py
class Solution:
    def triangleNumber(self, nums: List[int]) -> int:
        if not nums or len(nums) < 3:
            return 0
        
        nums.sort()
        ans = 0
        for maxIndex in range(2, len(nums)):
            ans += self.getTriangleCount(nums, nums[maxIndex], maxIndex)
        
        return ans
    
    def getTriangleCount(self, nums, target, maxIndex):
        left, right = 0, maxIndex - 1
        cnt = 0
        while left < right:
            if nums[left] + nums[right] > target:
                cnt += right - left
                right -= 1
            else:
                left += 1
        
        return cnt
        

File: Problems/Two Sum II - Input array is sorted/mine.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n = numbers.size();
        int left = 0, right = n - 1;
        vector<int> ans;
        while (true){
            int temp = numbers[left] + numbers[right];
            if (temp == target){
                ans.push_back(left + 1);
                ans.push_back(right + 1);
                break;
            }
            else if (temp > target){
                right--;
            }
            else{
                left++;
            }
        }
        return ans;
    }
};
